### PR TITLE
AMO 1690 flowsheet

### PR DESCRIPTION
**Pull request by tarnold17**
## Fixes/Resolves:

Add flowsheet for AMO 1690.

## Changes proposed in this PR:
-
-

### Legal Acknowledgement

By contributing to this software project, I agree to the following terms and conditions for my contribution:

1. I agree my contributions are submitted under the license terms described in the LICENSE.txt file at the top level of this directory.
2. I represent I am authorized to make the contributions and grant the license. If my employer has rights to intellectual property that includes these contributions, I represent that I have received permission to make contributions and grant the required license on behalf of that employer.

### CODE REVIEW

**Comment by adam-a-a** [comment]
@tarnold17 I see that the mac test is failing -- this typically isn't an issue for ZO model flowsheets, so not sure what the issue is yet, but we usually add a pytest mark to skip any test that depends on a solve (to skip mac test).

File: watertap/examples/flowsheets/case_studies/wastewater_resource_recovery/amo_1690/amo_1690.py [hunk 1]
```diff
@@ -0,0 +1,582 @@
...
+        )
+    )
+    print(f"Biogas production: {biogas_production:.2f} m^3/hr")
+
+    byproduct_tss_rate = value(
+        pyunits.convert(
+            m.fs.ad_byproduct.flow_mass_comp[0, "tss"], to_units=pyunits.kg / pyunits.hr
+        )
+    )
+    print(f"TSS byproduct production: {byproduct_tss_rate:.2f} m^3/hr")
```
Comment (MarcusHolly) [inline watertap/examples/flowsheets/case_studies/wastewater_resource_recovery/amo_1690/amo_1690.py]:
```suggestion
    print(f"TSS byproduct production: {byproduct_tss_rate:.2f} kg/hr")
```

File: watertap/examples/flowsheets/case_studies/wastewater_resource_recovery/amo_1690/amo_1690.py [hunk 2]
```diff
@@ -0,0 +1,582 @@
...
+
+    byproduct_cod_rate = value(
+        pyunits.convert(
+            m.fs.ad_byproduct.flow_mass_comp[0, "cod"], to_units=pyunits.kg / pyunits.hr
+        )
+    )
+    print(f"COD byproduct production: {byproduct_cod_rate:.2f} m^3/hr")
```
Comment (MarcusHolly) [inline watertap/examples/flowsheets/case_studies/wastewater_resource_recovery/amo_1690/amo_1690.py]:
```suggestion
    print(f"COD byproduct production: {byproduct_cod_rate:.2f} kg/hr")
```

File: watertap/examples/flowsheets/case_studies/wastewater_resource_recovery/amo_1690/amo_1690.py [hunk 3]
```diff
@@ -0,0 +1,582 @@
...
+
+    byproduct_tkn_rate = value(
+        pyunits.convert(
+            m.fs.ad_byproduct.flow_mass_comp[0, "tkn"], to_units=pyunits.kg / pyunits.hr
+        )
+    )
+    print(f"TKN byproduct production: {byproduct_tkn_rate:.2f} m^3/hr")
```
Comment (MarcusHolly) [inline watertap/examples/flowsheets/case_studies/wastewater_resource_recovery/amo_1690/amo_1690.py]:
```suggestion
    print(f"TKN byproduct production: {byproduct_tkn_rate:.2f} kg/hr")
```

**Review by MarcusHolly — APPROVED** [review]
LGTM - just a few inconsistencies with the units (and the mac test failing). But it looks like you were able to get the FeedZO deprecation resolved, so thanks for handling that!

File: watertap/examples/flowsheets/case_studies/wastewater_resource_recovery/amo_1690/amo_1690.py (hunk 1 above)
Comment (tarnold17) [reply watertap/examples/flowsheets/case_studies/wastewater_resource_recovery/amo_1690/amo_1690.py]: Good eye finding these. Addressed in latest commit.

File: watertap/examples/flowsheets/case_studies/wastewater_resource_recovery/amo_1690/amo_1690_case_study.yaml [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+    units: 1/year
+  laboratory_fees_percent_FCI:
+    value: 0
+    units: 1/year
+  insurance_and_taxes_percent_FCI:
+    value: 0
+    units: 1/year
+  wacc:
+    # Weighted Average Cost of Capital (WACC)
+    value: 0.08
```
Comment (adam-a-a) [inline watertap/examples/flowsheets/case_studies/wastewater_resource_recovery/amo_1690/amo_1690_case_study.yaml:43]:
```suggestion
    value: 0.05
```

File: watertap/examples/flowsheets/case_studies/wastewater_resource_recovery/amo_1690/amo_1690_case_study.yaml [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+    units: USD_2020/kWh
+  biogas:
+    value: 0.4
+    units: USD_2020/m^3
+  fertilizer:
+    value: 0.5
+    units: USD_2020/kg
+global_parameters:
+  plant_lifetime:
+    value: 20
```
Comment (adam-a-a) [inline watertap/examples/flowsheets/case_studies/wastewater_resource_recovery/amo_1690/amo_1690_case_study.yaml:15]:
```suggestion
    value: 30
```
Not sure about this- just compared to a couple other global yaml files, trying to make sure this lines up with the majority of the other cases (could be some deviation for earlier ones where we skipped enforcing all values to be same across flowsheets)

File: watertap/examples/flowsheets/case_studies/wastewater_resource_recovery/amo_1690/amo_1690_case_study.yaml [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+    units: year
+  utilization_factor:
+    value: 0.9
```
Comment (adam-a-a) [inline watertap/examples/flowsheets/case_studies/wastewater_resource_recovery/amo_1690/amo_1690_case_study.yaml:18]:
```suggestion
    value: 0.85
```
Same comment as above

Comment (tarnold17) [reply watertap/examples/flowsheets/case_studies/wastewater_resource_recovery/amo_1690/amo_1690_case_study.yaml:18]: 0.85 is usually our default, but this team gave us some TEA/LCA data that lists 0.9. So I used that instead.

File: watertap/examples/flowsheets/case_studies/wastewater_resource_recovery/amo_1690/amo_1690_case_study.yaml (hunk 5 above)
Comment (tarnold17) [reply watertap/examples/flowsheets/case_studies/wastewater_resource_recovery/amo_1690/amo_1690_case_study.yaml:15]: 30 years is usually our default, but this team gave us some TEA/LCA data that lists 20. So I used that instead.

File: watertap/examples/flowsheets/case_studies/wastewater_resource_recovery/amo_1690/amo_1690_case_study.yaml (hunk 4 above)
Comment (tarnold17) [reply watertap/examples/flowsheets/case_studies/wastewater_resource_recovery/amo_1690/amo_1690_case_study.yaml:43]: This is just the discount rate, right? 0.05 is usually our default, but this team gave us some TEA/LCA data that lists 0.08. So I used that instead.

File: watertap/examples/flowsheets/case_studies/wastewater_resource_recovery/amo_1690/tests/test_amo_1690_sweep.py [hunk 7]
```diff
@@ -0,0 +1,44 @@
...
+import tempfile
+from watertap.examples.flowsheets.case_studies.wastewater_resource_recovery.amo_1690.amo_1690_sweep import (
+    run_analysis,
+)
+
+sweep_list = [
+    1,
+    2,
+    3,
+]
```
Comment (adam-a-a) [inline watertap/examples/flowsheets/case_studies/wastewater_resource_recovery/amo_1690/tests/test_amo_1690_sweep.py:25]: DO we need all 3 for the test?

**Review by adam-a-a — APPROVED** [review]
LGTM - any further modifications can be made on a subsequent PR, which may arise while putting together the results report.

File: watertap/examples/flowsheets/case_studies/wastewater_resource_recovery/amo_1690/tests/test_amo_1690_sweep.py (hunk 7 above)
Comment (tarnold17) [reply watertap/examples/flowsheets/case_studies/wastewater_resource_recovery/amo_1690/tests/test_amo_1690_sweep.py:25]: I suppose we don't "need' to test all three cases, but I think it's probably best practice to do so. The tests use nx=2, so it shouldn't take much additional time to test them all.

**Pull request merged**